### ISSUE TITLE
How to implement torch.nn.functional.grid_sample padding mode in Tensorflow?

### ISSUE THREAD

**Issue by Eurus-Holmes** [tensorflow/models#7380]
https://stackoverflow.com/questions/52888146/what-is-the-equivalent-of-torch-nn-functional-grid-sample-in-tensorflow-numpy

How to implement `padding mode`?

------------------------

### System information

- **TensorFlow version (use command below)**: 1.10.0

### Describe the problem
How to implement torch.nn.functional.grid_sample padding mode in Tensorflow?

### Source code / logs
https://github.com/tensorflow/models/blob/master/research/transformer/spatial_transformer.py
